### ISSUE TITLE
Username on card Shop

### ISSUE THREAD

**Issue by heltonlr** [cambiatus/frontend#103]
Muda users missed the name of the person selling a product in the Shop, just the photo was not enough. 

So the improvement is to add the first username on the cards...

**MOBILE**
<img width="386" alt="Captura de Tela 2019-12-09 às 09 56 51" src="https://user-images.githubusercontent.com/39558132/70437393-4bdf4380-1a6a-11ea-96b1-7b6294152abc.png">

<img width="682" alt="Captura de Tela 2019-12-09 às 10 16 03" src="https://user-images.githubusercontent.com/39558132/70438602-f9535680-1a6c-11ea-9bd0-9b5378723f5f.png">

**DESK**
<img width="322" alt="Captura de Tela 2019-12-09 às 09 56 37" src="https://user-images.githubusercontent.com/39558132/70437395-4c77da00-1a6a-11ea-807f-1eeddde88412.png">

<img width="660" alt="Captura de Tela 2019-12-09 às 10 16 16" src="https://user-images.githubusercontent.com/39558132/70438618-ffe1ce00-1a6c-11ea-9e2b-c9bcc7552b22.png">

**Interface attributes**

https://www.figma.com/file/idejYyPJDeoqJvodGUCoiV/Cards?node-id=0%3A1

and

https://www.figma.com/file/OMN5j9HUcRQwFbMcVqaBwi/Productpage?node-id=0%3A1

**Comment by namoscagnm** [comment]
Hi, can you assign this to me so I can have a try? It looks an interesting problem :)

**Comment by namoscagnm** [comment]
Hi, actually we can already see the user name when we click buy. Do you mean you want to see it before buy, right? There are two places where this could happen: While listing all products, or after clicking one in the list, but before clicking buy. Where do you need it to be? Both? Thanks

**Comment by namoscagnm** [comment]
Hi, I see the user account alias (shortname) is being shown, so I tried to see how it works. 

It starts (backwards) on Viewer.elm, line 662, that the user account is being loaded with:
accountName, which comes from card.sale.creatorId, line 615. I see fro Sale.elm, page 615, that creatorId is creatorId : Eos.Name, which comes from Eos.Account in file Account.elm, line 32.

I somehow got the impression that Eos.Account only registers user alias, but I think if I knew how to debug where values come from and how they are generated it would be easier for me. Do you have a useful link about tracing the Elm content generation (debugging)? Thanks!

**Comment by lucca65** [comment]
Done @namoscagnm you are assigned. 

Maybe you'll have to modify the Shop query to include more information about the user. You can do that on graphql first and then on elm. You can test and explore our api here: https://api.cambiatus.io/api/graphiql

an sample query that includes more information about the user:

```graphql
query {
  sales(input: { account: "lucca" }) {
    description
    price
    creator {
      name
    }
  }
}
```

**Comment by zacck** [comment]
@namoscagnm you can always use the small blue box on the right lower side of the screen to view the model you are operating on and what pieces of data you need to add let me know if I can assist more with this ...

**Comment by namoscagnm** [comment]
Can you correct me if I'm on the wrong track with those two items: 

- I guess I have to change line 36 on Shop.elm to creator: {name: Name, avatar: Avatar}, and salesSelection near line 115 to do the equivalent change?

- I dont think I have to change Sale.elm under BeSpiral.Object because its autogenerate with all the possible data

**Comment by lucca65** [comment]
no need to do nothing with the files under Bespiral/

**Comment by zacck** [comment]
@namoscagnm Actually you may be a little lost, don't worry though it is a huge app with quite a bit of code so misdirection can easily occur. However see below on how you can achieve this
1. The information you need is already fetched so good news you don't need to touch the shop module at all. 
2. You will need to adjust this area of code https://github.com/cambiatus/frontend/blob/5580193af833ecc3e30dc5041d53d2446f0b9f8d/src/elm/Page/Shop.elm#L335

What you can do is open the shop page and click on the blue box at the bottom right
![Screenshot 2019-12-19 at 07 32 30](https://user-images.githubusercontent.com/897731/71147663-1dafef80-2232-11ea-94ec-1f2201287176.png)

This will in turn show you the current model you are working on as below
![Screenshot 2019-12-19 at 07 33 20](https://user-images.githubusercontent.com/897731/71147687-30c2bf80-2232-11ea-8b8d-da929c4c6408.png)

From here you can see that the shop indeed has a list of cards and the list of cards has a creatorID
![Screenshot 2019-12-19 at 07 33 39](https://user-images.githubusercontent.com/897731/71147821-a038af00-2232-11ea-9b57-6a4d4afa05f0.png)


Critically this is the information you need as in you need to now start displaying the `card.sale.creatorId`

let me know if this is helpful

**Comment by namoscagnm** [comment]
Yes, there is alreadz the creatorId, but @heltonlr  asked for the username on his request. The creatorId is alredy shown anyway when you click buy, so I the request of this issue seems redundant.

**Comment by zacck** [comment]
Ah I see @namoscagnm so one thing you have to be aware of, the username isn't mandatory on this system so the user may or may not have a username perhaps you want to consider  the strategy employed in this part
https://github.com/cambiatus/frontend/blob/5580193af833ecc3e30dc5041d53d2446f0b9f8d/src/elm/Page/Dashboard/Community.elm#L439

basically you show a combination of the the username and the creator id, 

This is still valid  as the users would like to see the username/creatorid before they press the buy button

**Comment by lucca65** [comment]
Just a clarification folks. In the system we use EOS, that require an account. Accounts are always 12 character long strings. Besides that our users also have a Name, with a max length of 256 characters. 

What we need to do is start to display on the listing of sales and in the individual sale the name of the user. We will have scenarios where the name will be `""` or `NULL` but never a scenario where the account isn't there. 

It means that you'll have to check `sale.creator.name` and `sale.creatorId` and display those accordingly.

Also @namoscagnm, you will note that on the sale detail the user name only appears when you click the "buy" button. Check print 1 and 2 for comparison.

You'll have to make sure that the username is always present, not only when showing the form.

# Prints

<img width="502" alt="Screen Shot 2019-12-19 at 12 39 04" src="https://user-images.githubusercontent.com/1082127/71186880-bd827300-225c-11ea-9f6f-44f1879e948d.png">
<img width="504" alt="Screen Shot 2019-12-19 at 12 39 13" src="https://user-images.githubusercontent.com/1082127/71186881-bd827300-225c-11ea-8cde-60414bd8dbe1.png">

**Comment by lucca65** [comment]
closed by #121
